starter-analytics-postgres/src/main/java/io/micronaut/starter/analytics/postgres/ExcelGenerator.java

```diff
@@ -0,0 +1,128 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.analytics.postgres;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.options.JdkVersion;
+import jakarta.inject.Singleton;
+import org.apache.poi.ss.usermodel.*;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Generates Excel spreadsheets.
+ */
+@Singleton
+public class ExcelGenerator {
+
+    private static final String SHEET_NAME = "Applications";
+
+    private static final Function<Workbook, CellStyle> CELL_STYLE_DATE_TIME = workbook -> {
+        final CellStyle style = workbook.createCellStyle();
+        style.setDataFormat((short) BuiltinFormats.getBuiltinFormat("m/d/yy h:mm"));
+        return style;
+    };
+
+    private static final Function<Workbook, CellStyle> CELL_STYLE_NULL = workbook -> null;
+
+    /**
+     * Generates an Excel spreadsheet containing all the given applications.
+     *
+     * @return An Excel spreadsheet.
+     */
+    public Workbook generateExcel(@NonNull Collection<Application> applications) {
+        final Workbook workbook = new XSSFWorkbook();
+        final Sheet sheet = workbook.createSheet(SHEET_NAME);
+        int rownum = 0;
+        // Headers
+        final CellStyle locked = workbook.createCellStyle();
+        locked.setLocked(true);
+        final Row headers = sheet.createRow(rownum++);
+        headers.setRowStyle(locked);
+        for (ExcelColumn column : ExcelColumn.values()) {
+            headers.createCell(column.ordinal(), CellType.STRING).setCellValue(column.title);
+        }
+        // Data
+        for (Application app : applications) {
+            Row data = sheet.createRow(rownum++);
+            for (ExcelColumn column : ExcelColumn.values()) {
+                final Cell cell = data.createCell(column.ordinal(), column.type);
+                column.setter.accept(app, cell);
+                Optional.ofNullable(column.styler.apply(workbook)).ifPresent(cell::setCellStyle);
+            }
+        }
+        return workbook;
+    }
+
+    private enum ExcelColumn {
+        ID("id", CellType.NUMERIC, numeric(Application::getId)),
+        FEATURES("features", CellType.STRING, features(Application::getFeatures)),
```

Comment:
column order should be 

```
date_created
type
language
build_tool
test_framework
jdk_version
micronaut_version
id
features
```

Reply:
[Issue description](https://github.com/micronaut-projects/micronaut-starter/issues/1976#issue-1817886739) specified different order and case:

> it should have a header row
>
> `id | features| type | language | buildTool | testFramework | jdkVersion | micronautVersion | dateCreated`

@sdelamo  Can you please confirm which one is the correct one?

Reply:
I changed my mind from the original definition. I want features to be last in case I want to expand it immediately after